packages/theme-addresses/__tests__/theme-addresses.test.js

```diff
@@ -0,0 +1,132 @@
+
+const shippingCountryOptions = '<option value="---" data-provinces="[]">---</option>' +
+  '<option value="France" data-provinces="[]">France</option>' +
+  '<option value="United Kingdom" data-provinces="[]">United Kingdom</option>';
+
+describe('CountryProvinceSelector', () => {
+  test('is a function exported by theme-addresses.js', () => {
+    expect(typeof CountryProvinceSelector).toBe('function');
+  });
+
+  test('returns an object with a build function', () => {
+    let countryProvinceSelector;
+    expect(() => {
+      countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    }).not.toThrow();
+    expect(typeof countryProvinceSelector.build).toBe('function');
+  });
+
+  // Hmm ... how to validate the countryOptions that is being passed in is a valid html?
+});
+
+describe('CountryProvinceSelector.build()', () => {
+  test('populates the given country selectors when no default is set', () => {
+    document.body.innerHTML = `<form>
+      <select id="addressTestCountry"></select>
+      <select id="addressTestProvince"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#addressTestCountry');
+    const provinceSelector = document.querySelector('#addressTestProvince');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector);
+
+    expect(countrySelector.querySelectorAll('option').length).not.toEqual(0);
+    expect(countrySelector.value).toEqual('---');
+    expect(countrySelector.querySelectorAll('option[value="Canada"]').length).toEqual(1);
+  });
+
+  test('populates the given country and province selectors when a default is set for country', () => {
+    document.body.innerHTML = `<form>
+      <select id="addressTestCountry" data-default="New Zealand"></select>
+      <select id="addressTestProvince"></select>
+    </form>`;
+
+    const countryProvinceSelector = new CountryProvinceSelector(countryOptions);
+    const countrySelector = document.querySelector('#addressTestCountry');
+    const provinceSelector = document.querySelector('#addressTestProvince');
+  
+    countryProvinceSelector.build(countrySelector, provinceSelector);
+
+    expect(countrySelector.value).toEqual('New Zealand');
+    expect(provinceSelector.querySelectorAll('option').length).not.toEqual(0);
+    expect(provinceSelector.value).toEqual('---');
+    expect(provinceSelector.querySelectorAll('option[value="Quebec"]').length).toEqual(1);
```

Comment:
There is no Quebec in New Zealand. This should be failing.

Reply:
Yes .. which means the DOM might not be clean as I thought

Reply:
Oh wait, never mind. The test failed at the step before anyways.